### DISCUSSION TITLE
Example of file monitoring/auto restarting

### DISCUSSION THREAD

**Discussion by smolattack** [Badgerati/Pode#1124] [Q&A]
So how do I set up my project so that the server restarts whenever I make a change so that I don't need to do it manually each time? The documentation doesn't have an example of how that would work at time of writing this.

**Comment by Badgerati** [comment]
Hi @smolattack,

Do you mean [this](https://badgerati.github.io/Pode/Tutorials/Restarting/Types/FileMonitoring/)? Or something different?

If you mean when you make a change to the server.ps1 and the contents of the `Start-PodeServer` scriptblock itself, then this isn't possible. To allow updates to be reloaded you'll have to dot source in files with the required content, or use the `Use-PodeRoutes` etc. functions.

**Reply by smolattack** [reply]
Yes that's what I mean. Please provide an example of dot-sourcing a route or an example showing how to use Use-PodeRoutes

**Reply by Badgerati** [reply]
I thought there was until I looked! The function summary on `Use-PodeRoutes` does give a description/example of it's use, but the `Use-PodeX` functions do appear to all be missing, other than getting a mention like [here](https://badgerati.github.io/Pode/Tutorials/Routes/Overview/#usage).

**Reply by smolattack** [reply]
Ok thanks that worked but was very confusing.

The section with the example is called Script from File and its link is below.
https://badgerati.github.io/Pode/Tutorials/Schedules/#script-from-file

The confusing part is that there is an identically named section in the Basics page (where you might think to initially look for this information in the first place)
https://badgerati.github.io/Pode/Tutorials/Basics/#script-from-file
It's not clear then where you put your `import-module -Name Pode` (in server.ps1 or File.ps1). I've never put the whole contents of a script file in a script block before. Would probably help if things like that were explained and have a dedicated DX page where all this information can easily be found.

I kept ending up in the "File Watcher" guide. The welcome page gave me a hint to search for "file monitoring" in one of the bullet points, which takes you to the DX guide (whereas File Watcher takes you to the API guide).

On a side note, I found a way to debug the server in one of the issues. Would be nice if there was a debug guide in the docs as well (or as a section in a DX page).
